[Bug]: Build Failed `carbonyl-host-tools`

### Problem description

[1630/16059] CXX host/obj/third_party/perfetto/src/tracing/client_api_without_backends/track_event_internal.o
FAILED: host/obj/third_party/perfetto/src/tracing/client_api_without_backends/track_event_internal.o 
/usr/lib/llvm-19/bin/clang++ -MMD -MF host/obj/third_party/perfetto/src/tracing/client_api_without_backends/track_event_internal.o.d -DUSE_AURA=1 -DUSE_OZONE=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_GNU_SOURCE -DCR_SYSROOT_KEY=20221105T211506Z-0 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DPERFETTO_IMPLEMENTATION -I../../../src/chromium/src -Ihost/gen -I../../../src/chromium/src/third_party/perfetto -I../../../src/chromium/src/third_party/perfetto/src/profiling/memory/include -I../../../src/chromium/src/third_party/perfetto/include -Ihost/gen/third_party/perfetto/build_config -Ihost/gen/third_party/perfetto -I../../../src/chromium/src/third_party/protobuf/src -Ihost/gen/protoc_out -Wall -Wextra -Wimplicit-fallthrough -Wextra-semi -Wunreachable-code-aggressive -Wthread-safety -Wno-missing-field-initializers -Wno-unused-parameter -Wno-psabi -Wno-unknown-warning-option -Wno-unknown-pragmas -Wloop-analysis -
In file included from ../../../src/chromium/src/third_party/perfetto/src/tracing/internal/track_event_internal.cc:23:
In file included from ../../../src/chromium/src/third_party/perfetto/include/perfetto/tracing/track_event.h:21:
../../../src/chromium/src/third_party/perfetto/include/perfetto/tracing/internal/track_event_data_source.h:254:20: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  254 |     Base::template Trace([](typename Base::TraceContext ctx) { ctx.Flush(); });
      |                    ^
../../../src/chromium/src/third_party/perfetto/include/perfetto/tracing/internal/track_event_data_source.h:260:20: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  260 |     Base::template CallIfEnabled(
      |                    ^
../../../src/chromium/src/third_party/perfetto/include/perfetto/tracing/internal/track_event_data_source.h:275:20: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  275 |     Base::template Trace([&](typename Base::TraceContext ctx) {
      |                    ^
../../../src/chromium/src/third_party/perfetto/include/perfetto/tracing/internal/track_event_data_source.h:449:20: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  449 |     Base::template Trace([&](typename Base::TraceContext ctx) {
      |                    ^
../../../src/chromium/src/third_party/perfetto/include/perfetto/tracing/internal/track_event_data_source.h:611:22: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  611 |       Base::template TraceWithInstances(instances, std::move(lambda));
      |                      ^
../../../src/chromium/src/third_party/perfetto/include/perfetto/tracing/internal/track_event_data_source.h:625:20: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  625 |     Base::template Trace([&](typename Base::TraceContext ctx) {
      |                    ^
6 errors generated.
[1631/16059] CXX host/obj/third_party/perfetto/src/tracing/client_api_without_backends/traced_value.o
[1632/16059] CXX host/obj/third_party/perfetto/src/tracing/client_api_without_backends/tracing_policy.o
[1633/16059] CXX host/obj/third_party/perfetto/src/tracing/client_api_without_backends/tracing.o
[1634/16059] CXX host/obj/third_party/perfetto/src/tracing/client_api_without_backends/tracing_muxer_impl.o
[1635/16059] CXX host/obj/third_party/perfetto/src/trace_processor/db/db/column.o
ninja: build stopped: subcommand failed.
real	9m21.171s
user	33m30.622s
sys	1m56.639s
Error: Process completed with exit code 1.

### What steps will reproduce the bug?

build

### What is the expected behavior?

_No response_

### System information

```shell
NA
```

--- a/x11-packages/carbonyl-host-tools/5001-clang-19.patch
+++ b/x11-packages/carbonyl-host-tools/5001-clang-19.patch
@@ -1,0 +1,87 @@
+https://github.com/termux/termux-packages/issues/25308
+
+--- a/chromium/src/third_party/perfetto/include/perfetto/tracing/internal/track_event_data_source.h
++++ b/chromium/src/third_party/perfetto/include/perfetto/tracing/internal/track_event_data_source.h
+@@ -251,13 +251,13 @@
+   }
+ 
+   static void Flush() {
+-    Base::template Trace([](typename Base::TraceContext ctx) { ctx.Flush(); });
++    Base::Trace([](typename Base::TraceContext ctx) { ctx.Flush(); });
+   }
+ 
+   // Determine if *any* tracing category is enabled.
+   static bool IsEnabled() {
+     bool enabled = false;
+-    Base::template CallIfEnabled(
++    Base::CallIfEnabled(
+         [&](uint32_t /*instances*/) { enabled = true; });
+     return enabled;
+   }
+@@ -272,7 +272,7 @@
+   static bool IsDynamicCategoryEnabled(
+       const DynamicCategory& dynamic_category) {
+     bool enabled = false;
+-    Base::template Trace([&](typename Base::TraceContext ctx) {
++    Base::Trace([&](typename Base::TraceContext ctx) {
+       enabled = IsDynamicCategoryEnabled(&ctx, dynamic_category);
+     });
+     return enabled;
+@@ -446,7 +446,7 @@
+                                  const protos::gen::TrackDescriptor& desc) {
+     PERFETTO_DCHECK(track.uuid == desc.uuid());
+     TrackRegistry::Get()->UpdateTrack(track, desc.SerializeAsString());
+-    Base::template Trace([&](typename Base::TraceContext ctx) {
++    Base::Trace([&](typename Base::TraceContext ctx) {
+       TrackEventInternal::WriteTrackDescriptor(
+           track, ctx.tls_inst_->trace_writer.get(), ctx.GetIncrementalState(),
+           *ctx.GetCustomTlsState(), TrackEventInternal::GetTraceTime());
+@@ -608,7 +608,7 @@
+                                  Lambda lambda) PERFETTO_ALWAYS_INLINE {
+     using CatTraits = CategoryTraits<CategoryType>;
+     if (CatTraits::kIsDynamic) {
+-      Base::template TraceWithInstances(instances, std::move(lambda));
++      Base::TraceWithInstances(instances, std::move(lambda));
+     } else {
+       Base::template TraceWithInstances<CategoryTracePointTraits>(
+           instances, std::move(lambda), {CatTraits::GetStaticIndex(category)});
+@@ -622,7 +622,7 @@
+       const TrackType& track,
+       std::function<void(protos::pbzero::TrackDescriptor*)> callback) {
+     TrackRegistry::Get()->UpdateTrack(track, std::move(callback));
+-    Base::template Trace([&](typename Base::TraceContext ctx) {
++    Base::Trace([&](typename Base::TraceContext ctx) {
+       TrackEventInternal::WriteTrackDescriptor(
+           track, ctx.tls_inst_->trace_writer.get(), ctx.GetIncrementalState(),
+           *ctx.GetCustomTlsState(), TrackEventInternal::GetTraceTime());
+--- a/chromium/src/third_party/blink/renderer/platform/wtf/hash_table.h
++++ b/chromium/src/third_party/blink/renderer/platform/wtf/hash_table.h
+@@ -1668,7 +1668,7 @@
+     }
+   }
+   table_ = temporary_table;
+-  Allocator::template BackingWriteBarrier(&table_);
++  Allocator::BackingWriteBarrier(&table_);
+ 
+   HashTableBucketInitializer<Traits, Allocator, Value>::InitializeTable(
+       original_table, new_table_size);
+@@ -1722,7 +1722,7 @@
+   // This swaps the newly allocated buffer with the current one. The store to
+   // the current table has to be atomic to prevent races with concurrent marker.
+   AsAtomicPtr(&table_)->store(new_hash_table.table_, std::memory_order_relaxed);
+-  Allocator::template BackingWriteBarrier(&table_);
++  Allocator::BackingWriteBarrier(&table_);
+   table_size_ = new_table_size;
+ 
+   new_hash_table.table_ = old_table;
+@@ -1874,8 +1874,8 @@
+   // on the mutator thread, which is also the only one that writes to them, so
+   // there is *no* risk of data races when reading.
+   AtomicWriteSwap(table_, other.table_);
+-  Allocator::template BackingWriteBarrier(&table_);
+-  Allocator::template BackingWriteBarrier(&other.table_);
++  Allocator::BackingWriteBarrier(&table_);
++  Allocator::BackingWriteBarrier(&other.table_);
+   if (IsWeak<ValueType>::value) {
+     // Weak processing is omitted when no backing store is present. In case such
+     // an empty table is later on used it needs to be strongified.

--- a/x11-packages/carbonyl-host-tools/5002-quiche-size-typo.patch
+++ b/x11-packages/carbonyl-host-tools/5002-quiche-size-typo.patch
@@ -1,0 +1,19 @@
+https://github.com/google/quiche/commit/c1af894e0f5c4f732a983e7c93227854e203570e
+
+--- a/chromium/src/net/third_party/quiche/src/quiche/quic/core/quic_interval_deque.h
++++ b/chromium/src/net/third_party/quiche/src/quiche/quic/core/quic_interval_deque.h
+@@ -198,12 +198,12 @@
+     Iterator operator+(difference_type amount) const {
+       Iterator copy = *this;
+       copy.index_ += amount;
+-      QUICHE_DCHECK(copy.index_ < copy.deque_->size());
++      QUICHE_DCHECK(copy.index_ < copy.deque_->Size());
+       return copy;
+     }
+     Iterator& operator+=(difference_type amount) {
+       index_ += amount;
+-      QUICHE_DCHECK(index_ < deque_->size());
++      QUICHE_DCHECK(index_ < deque_->Size());
+       return *this;
+     }
+     difference_type operator-(const Iterator& rhs) const {

--- a/x11-packages/carbonyl-host-tools/5003-base-containers-id_map-typo.patch
+++ b/x11-packages/carbonyl-host-tools/5003-base-containers-id_map-typo.patch
@@ -1,0 +1,15 @@
+https://github.com/chromium/chromium/commit/37ef38092ab783d0126922e8d463024341c481b9
+
+--- a/chromium/src/base/containers/id_map.h
++++ b/chromium/src/base/containers/id_map.h
+@@ -155,8 +155,8 @@
+     }
+ 
+     const Iterator& operator=(const Iterator& iter) {
+-      map_ = iter.map;
+-      iter_ = iter.iter;
++      map_ = iter.map_;
++      iter_ = iter.iter_;
+       Init();
+       return *this;
+     }

--- a/x11-packages/carbonyl-host-tools/build.sh
+++ b/x11-packages/carbonyl-host-tools/build.sh
@@ -5,11 +5,15 @@ TERMUX_PKG_LICENSE_FILE="license.md"
 TERMUX_PKG_MAINTAINER="@licy183"
 _CHROMIUM_VERSION=111.0.5563.146
 TERMUX_PKG_VERSION=0.0.3
-TERMUX_PKG_REVISION=1
-TERMUX_PKG_SRCURL=(https://github.com/fathyb/carbonyl/archive/refs/tags/v$TERMUX_PKG_VERSION.tar.gz
-					https://commondatastorage.googleapis.com/chromium-browser-official/chromium-$_CHROMIUM_VERSION.tar.xz)
-TERMUX_PKG_SHA256=(bf421b9498a084a7cf2238a574d37d31b498d3e271fdb3dcf466e7ed6c80013d
-					1e701fa31b55fa0633c307af8537b4dbf67e02d8cad1080c57d845ed8c48b5fe)
+TERMUX_PKG_REVISION=2
+TERMUX_PKG_SRCURL=(
+	https://github.com/fathyb/carbonyl/archive/refs/tags/v$TERMUX_PKG_VERSION.tar.gz
+	https://commondatastorage.googleapis.com/chromium-browser-official/chromium-$_CHROMIUM_VERSION.tar.xz
+)
+TERMUX_PKG_SHA256=(
+	bf421b9498a084a7cf2238a574d37d31b498d3e271fdb3dcf466e7ed6c80013d
+	1e701fa31b55fa0633c307af8537b4dbf67e02d8cad1080c57d845ed8c48b5fe
+)
 TERMUX_PKG_DEPENDS="atk, cups, dbus, fontconfig, gtk3, krb5, libc++, libdrm, libevdev, libxkbcommon, libminizip, libnss, libwayland, libx11, mesa, openssl, pango, pulseaudio, zlib"
 TERMUX_PKG_BUILD_DEPENDS="libnotify, libffi-static"
 TERMUX_PKG_ON_DEVICE_BUILD_NOT_SUPPORTED=true
@@ -26,6 +30,7 @@ termux_step_post_get_source() {
 	# Move chromium source
 	mv chromium-$_CHROMIUM_VERSION chromium/src
 
+	local f
 	# Apply diffs
 	for f in $(find "$TERMUX_PKG_BUILDER_DIR/" -maxdepth 1 -type f -name *.diff | sort); do
 		echo "Applying patch: $(basename $f)"
@@ -244,6 +249,11 @@ use_jumbo_build = true
 }
 
 termux_step_make() {
+	# # Build libcarbonyl
+	# (cd $TERMUX_PKG_SRCDIR
+	# termux_setup_rust
+	# cargo build --jobs $TERMUX_PKG_MAKE_PROCESSES --target $CARGO_TARGET_NAME --release)
+
 	cd $TERMUX_PKG_BUILDDIR
 	# Build v8 snapshot and tools
 	time ninja -C out/Release \
@@ -266,6 +276,9 @@ termux_step_make() {
 	time ninja -C out/Release \
 						third_party/swiftshader/src/Vulkan:icd_file \
 						third_party/swiftshader/src/Vulkan:swiftshader_libvulkan
+
+	# # Build headless shell
+	# time ninja -C $TERMUX_PKG_BUILDDIR/out/Release headless:headless_shell
 }
 
 termux_step_make_install() {

